mlkem/native/x86_64/src/compress_avx2.c

```diff
@@ -0,0 +1,370 @@
+    _mm_storel_epi64((__m128i *)&r[22 * i + 16], t1);
+  }
+
+  f0 = _mm256_load_si256(&a[i]);
+  f1 = _mm256_mullo_epi16(f0, v8);
+  f2 = _mm256_add_epi16(f0, off);
+  f0 = _mm256_slli_epi16(f0, 3);
+  f0 = _mm256_mulhi_epi16(f0, v);
+  f2 = _mm256_sub_epi16(f1, f2);
+  f1 = _mm256_andnot_si256(f1, f2);
+  f1 = _mm256_srli_epi16(f1, 15);
+  f0 = _mm256_sub_epi16(f0, f1);
+  f0 = _mm256_mulhrs_epi16(f0, shift1);
+  f0 = _mm256_and_si256(f0, mask);
+  f0 = _mm256_madd_epi16(f0, shift2);
+  f0 = _mm256_sllv_epi32(f0, sllvdidx);
+  f1 = _mm256_bsrli_epi128(f0, 8);
+  f0 = _mm256_srlv_epi64(f0, srlvqidx);
+  f1 = _mm256_slli_epi64(f1, 34);
+  f0 = _mm256_add_epi64(f0, f1);
+  f0 = _mm256_shuffle_epi8(f0, shufbidx);
+  t0 = _mm256_castsi256_si128(f0);
+  t1 = _mm256_extracti128_si256(f0, 1);
+  t0 = _mm_blendv_epi8(t0, t1, _mm256_castsi256_si128(shufbidx));
+  _mm_storeu_si128((__m128i *)&r[22 * i + 0], t0);
+  /* Handle store in last iteration especially to avoid overflow */
+  memcpy(&r[22 * i + 16], &t1, 6);
+}
+
+void poly_decompress_d11_avx2(__m256i *RESTRICT r,
+                              const uint8_t a[MLKEM_POLYCOMPRESSEDBYTES_D11])
+{
+  unsigned int i;
+  __m256i f;
+  const __m256i q =
+      _mm256_load_si256(&qdata.vec[AVX2_BACKEND_DATA_OFFSET_16XQ / 16]);
+  const __m256i shufbidx =
+      _mm256_set_epi8(13, 12, 12, 11, 10, 9, 9, 8, 8, 7, 6, 5, 5, 4, 4, 3, 10,
+                      9, 9, 8, 7, 6, 6, 5, 5, 4, 3, 2, 2, 1, 1, 0);
+  const __m256i srlvdidx = _mm256_set_epi32(0, 0, 1, 0, 0, 0, 1, 0);
+  const __m256i srlvqidx = _mm256_set_epi64x(2, 0, 2, 0);
+  const __m256i shift =
+      _mm256_set_epi16(4, 32, 1, 8, 32, 1, 4, 32, 4, 32, 1, 8, 32, 1, 4, 32);
+  const __m256i mask = _mm256_set1_epi16(32752);
+
+  for (i = 0; i < (MLKEM_N / 16) - 1; i++)
+  {
+    f = _mm256_loadu_si256((__m256i *)&a[22 * i]);
+    f = _mm256_permute4x64_epi64(f, 0x94);
+    f = _mm256_shuffle_epi8(f, shufbidx);
+    f = _mm256_srlv_epi32(f, srlvdidx);
+    f = _mm256_srlv_epi64(f, srlvqidx);
+    f = _mm256_mullo_epi16(f, shift);
+    f = _mm256_srli_epi16(f, 1);
+    f = _mm256_and_si256(f, mask);
+    f = _mm256_mulhrs_epi16(f, q);
+    _mm256_store_si256(&r[i], f);
+  }
+
+  /* Handle load of last iteration especially */
+  memcpy(&f, &a[22 * i], 22);
+  /* The rest of the iteration is the same */
+  f = _mm256_permute4x64_epi64(f, 0x94);
+  f = _mm256_shuffle_epi8(f, shufbidx);
+  f = _mm256_srlv_epi32(f, srlvdidx);
+  f = _mm256_srlv_epi64(f, srlvqidx);
+  f = _mm256_mullo_epi16(f, shift);
+  f = _mm256_srli_epi16(f, 1);
+  f = _mm256_and_si256(f, mask);
+  f = _mm256_mulhrs_epi16(f, q);
+  _mm256_store_si256(&r[i], f);
+}
+
+#endif /* MLKEM_NATIVE_MULTILEVEL_BUILD || MLKEM_K == 4 */
+
+#else /* MLKEM_USE_NATIVE_X86_64 && SYS_X86_64_AVX2 */
+
+/* Dummy declaration for compilers disliking empty compilation units */
+#define empty_cu_avx2_polyvec_compress \
+  MLKEM_NAMESPACE(empty_cu_avx2_polyvec_compress)
+int empty_cu_avx2_polyvec_compress;
```

Comment:
Should this be changed in the same way as https://github.com/pq-code-package/mlkem-native/pull/706?

Reply:
Yes, fixed